Sources/Subscription/SubscriptionManager.swift

```diff
@@ -0,0 +1,132 @@
+    var subscriptionService: SubscriptionService { get }
+    var authService: AuthService { get }
+    @available(macOS 12.0, iOS 15.0, *)
+    func getStorePurchaseManager() -> StorePurchaseManaging
+    var currentEnvironment: SubscriptionEnvironment { get }
+    var canPurchase: Bool { get }
+    func loadInitialData()
+    func updateSubscriptionStatus(completion: @escaping (_ isActive: Bool) -> Void)
+}
+
+/// Single entry point for everything related to Subscription. This manager is disposable, every time something related to the environment changes this need to be recreated.
+final public class SubscriptionManager: SubscriptionManaging {
+
+    private let storePurchaseManager: StorePurchaseManaging?
+
+    public let accountManager: AccountManaging
+    public let subscriptionService: SubscriptionService
+    public let authService: AuthService
+    public let currentEnvironment: SubscriptionEnvironment
+    public private(set) var canPurchase: Bool = false
+
+    public init(storePurchaseManager: StorePurchaseManaging? = nil,
+                accountManager: AccountManaging,
+                subscriptionService: SubscriptionService,
+                authService: AuthService,
+                subscriptionEnvironment: SubscriptionEnvironment) {
+        self.storePurchaseManager = storePurchaseManager
+        self.accountManager = accountManager
+        self.subscriptionService = subscriptionService
+        self.authService = authService
+        self.currentEnvironment = subscriptionEnvironment
+        switch currentEnvironment.platform {
+        case .appStore:
+            if #available(macOS 12.0, iOS 15.0, *) {
+                setupForAppStore()
+            } else {
+                assertionFailure("Trying to setup AppStore where not supported")
+            }
+        case .stripe:
+            setupForStripe()
+        }
+    }
+
+    @available(macOS 12.0, iOS 15.0, *) public func getStorePurchaseManager() -> StorePurchaseManaging {
+        return storePurchaseManager!
+    }
+
+    // MARK: Load and Save SubscriptionEnvironment
+
+    static private let subscriptionEnvironmentStorageKey = "com.duckduckgo.subscription.environment"
+    static public func loadEnvironmentFrom(userDefaults: UserDefaults) -> SubscriptionEnvironment? {
+        if let savedData = userDefaults.object(forKey: Self.subscriptionEnvironmentStorageKey) as? Data {
+            let decoder = JSONDecoder()
+            if let loadedData = try? decoder.decode(SubscriptionEnvironment.self, from: savedData) {
+                return loadedData
+            }
+        }
+        return nil
+    }
+
+    static public func save(subscriptionEnvironment: SubscriptionEnvironment, userDefaults: UserDefaults) {
+        let encoder = JSONEncoder()
+        if let encodedData = try? encoder.encode(subscriptionEnvironment) {
+            userDefaults.set(encodedData, forKey: Self.subscriptionEnvironmentStorageKey)
+        }
+    }
```

Comment:
Not sure if for now, but have you thought about removing it from the `SubscriptionManager` altogether, tbt all it requires it is to initialize it with the current value of `SubscriptionEnvironment`. How and where it is persisted or what is the current vs default logic I see more of client's responsibility, wdyt?

Reply:
Not sure, the UserDefaults key is shared and the logic is shared between platforms and app extensions, why would you duplicate the code? I'm ok with moving it outside SubscriptionManager into a separate entity, I was thinking of doing it as part of https://app.asana.com/0/0/1207157941206688/f

Reply:
Let's leave it for now and eventually decide in the mentioned task later.

Reply:
A not for the future from usage observation. What about decoupling the `SubscriptionEnvironment` from the manager. Make `SubscriptionEnvironment` with an init that takes the `UserDefaults` and just handles the loading on init and saving on property changes.